casper/src/main/scala/coop/rchain/casper/engine/CasperLaunch.scala

```diff
@@ -40,170 +40,184 @@ object CasperLaunch {
     /* Casper */      : Estimator: SafetyOracle: LastFinalizedBlockCalculator: LastFinalizedHeightConstraintChecker: SynchronyConstraintChecker
     /* Storage */     : BlockStore: BlockDagStorage: LastFinalizedStorage: DeployStorage: CasperBufferStorage: RSpaceStateManager
     /* Diagnostics */ : Log: EventLog: Metrics: Span] // format: on
-  (conf: CasperConf): CasperLaunch[F] = new CasperLaunch[F] {
-    def launch(): F[Unit] =
-      BlockStore[F].getApprovedBlock map {
-        case Some(approvedBlock) =>
-          val msg    = "Approved block found, reconnecting to existing network"
-          val action = connectToExistingNetwork(approvedBlock)
-          (msg, action)
-        case None if (conf.genesisCeremony.genesisValidatorMode) =>
-          val msg =
-            "Approved block not found, taking part in ceremony as genesis validator"
-          val action = connectAsGenesisValidator()
-          (msg, action)
-        case None if (conf.genesisCeremony.ceremonyMasterMode) =>
-          val msg =
-            "Approved block not found, taking part in ceremony as ceremony master"
-          val action = initBootstrap()
-          (msg, action)
-        case None =>
-          val msg    = "Approved block not found, connecting to existing network"
-          val action = connectAndQueryApprovedBlock(conf.genesisCeremony.fromLastFinalizedState)
-          (msg, action)
-      } >>= {
-        case (msg, action) => Log[F].info(msg) >> action
-      }
+  (conf: CasperConf, allowRespondLastFinalizedStateRequest: Boolean): CasperLaunch[F] =
+    new CasperLaunch[F] {
+      def launch(): F[Unit] =
+        BlockStore[F].getApprovedBlock map {
+          case Some(approvedBlock) =>
+            val msg = "Approved block found, reconnecting to existing network"
+            val action =
+              connectToExistingNetwork(approvedBlock, allowRespondLastFinalizedStateRequest)
+            (msg, action)
+          case None if (conf.genesisCeremony.genesisValidatorMode) =>
+            val msg =
+              "Approved block not found, taking part in ceremony as genesis validator"
+            val action = connectAsGenesisValidator()
+            (msg, action)
+          case None if (conf.genesisCeremony.ceremonyMasterMode) =>
+            val msg =
+              "Approved block not found, taking part in ceremony as ceremony master"
+            val action = initBootstrap(allowRespondLastFinalizedStateRequest)
```

Comment:
I've mentioned this in another comment, but IMO CM have to respond to state transfer requests

Reply:
What is CM?

Reply:
@zsluedem Ceremony Master

Reply:
Oh nvm